### ISSUE TITLE
Tests should use VCR or a fixture server and avoid connecting to third-party services

### ISSUE THREAD

**Issue by Mr0grog** [edgi-govdata-archiving/web-monitoring-processing#236]
Some tests connect to third-party services, which we should avoid because they could cause intermittent and confusing failures in CI (like this one for @vbanos: https://github.com/edgi-govdata-archiving/web-monitoring-processing/pull/233#issuecomment-412326680). This also prevents tests from being run locally without a network connection, which is also no fun.

Our `web_monitoring.db` tests use VCR for this: https://github.com/edgi-govdata-archiving/web-monitoring-processing/blob/70292c8146b26ca9b82a0ae9e54fbd0e5db830df/web_monitoring/tests/test_db.py#L14-L59

And our `utils` tests use mock requests for simpler cases: https://github.com/edgi-govdata-archiving/web-monitoring-processing/blob/edc47ba9920b64e1381f1cec5adc7982bbf10f59/web_monitoring/tests/test_utils.py#L48-L53

First, we need to see what tests this includes. So far:

- [ ] Lots of these in [`test_diffing_server_exc_handling.py`](https://github.com/edgi-govdata-archiving/web-monitoring-processing/blob/master/web_monitoring/tests/test_diffing_server_exc_handling.py)

**Comment by Mr0grog** [comment]
Additionally, we need to be better about making sure future PRs that add new tests do this, too.

**Comment by Mr0grog** [comment]
More case in point: https://circleci.com/gh/edgi-govdata-archiving/web-monitoring-processing/585

**Comment by Mr0grog** [comment]
And again, this prevented @jsnshrmn from doing a release today: https://circleci.com/gh/edgi-govdata-archiving/web-monitoring-processing/772

**Comment by Mr0grog** [comment]
Update: need to re-check this issue for whether it still applies to anything in this repo after #638.

**Comment by Mr0grog** [comment]
This is indeed no longer relevant.
